src/AbstractNavItem.js

```diff
@@ -15,76 +16,68 @@ const propTypes = {
   onclick: PropTypes.func,
 
   as: PropTypes.any,
-
-  onSelect: PropTypes.func,
-
   onClick: PropTypes.func,
+  onSelect: PropTypes.func,
 };
 
 const defaultProps = {
   disabled: false,
 };
 
-class AbstractNavItem extends React.Component {
-  render() {
-    const {
+const AbstractNavItem = React.forwardRef(
+  (
+    {
       active,
       className,
       tabIndex,
       eventKey,
       onSelect,
+      onClick,
       as: Component,
       ...props
-    } = this.props;
-
+    },
+    ref,
+  ) => {
     const navKey = makeEventKey(eventKey, props.href);
+    const parentOnSelect = useContext(SelectableContext);
+    const navContext = useContext(NavContext);
+
+    let isActive = active;
+    if (navContext) {
+      if (!props.role && navContext.role === 'tablist') props.role = 'tab';
+
+      props['data-rb-event-key'] = navKey;
+      props.id = navContext.getControllerId(navKey);
+      props['aria-controls'] = navContext.getControlledId(navKey);
+
+      isActive =
+        active == null && navKey != null
+          ? navContext.activeKey === navKey
+          : active;
+    }
+
+    if (props.role === 'tab') {
+      props.tabIndex = isActive ? tabIndex : -1;
+      props['aria-selected'] = isActive;
+    }
+
+    const handleOnclick = useEventCallback(e => {
```

Comment:
for my benefit... what does `useEventCallback` do? i see the `useCommittedRef` thing but i don't immediately understand it

Reply:
it returns a function with a stable identity that doesn't need to be invalidated or changed if the inner handler isn't stable. it's a less flexible equivalent to 

```js
someMethod() {
  this.props.handleFoo()
}
```

Reply:
neat/complicated!